src/model/core/discharge/discharge.jl

```diff
@@ -35,15 +35,20 @@ function discharge!(EP::Model, inputs::Dict, setup::Dict)
 	@expression(EP, eTotalCVarOut, sum(eTotalCVarOutT[t] for t in 1:T))
 
 	# Add total variable discharging cost contribution to the objective function
-	EP[:eObj] += eTotalCVarOut
+	# EP[:eObj] += eTotalCVarOut
+	add_to_expression!(EP[:eObj], eTotalCVarOut)
 
 	# ESR Policy
 	if setup["EnergyShareRequirement"] >= 1
 
 		@expression(EP, eESRDischarge[ESR=1:inputs["nESR"]], sum(inputs["omega"][t]*dfGen[y,Symbol("ESR_$ESR")]*EP[:vP][y,t] for y=dfGen[findall(x->x>0,dfGen[!,Symbol("ESR_$ESR")]),:R_ID], t=1:T)
 						- sum(inputs["dfESR"][z,ESR]*inputs["omega"][t]*inputs["pD"][t,z] for t=1:T, z=findall(x->x>0,inputs["dfESR"][:,ESR])))
 
-		EP[:eESR] += eESRDischarge
+
+		for i=1:inputs["nESR"]
+			add_to_expression!(EP[:eESR][i], eESRDischarge[i])
+		end
+		# EP[:eESR] += eESRDischarge
```

Comment:
Can you say something about why the for loop is needed here?

Reply:
Ah, it looks like add_to_expression! only works on single values, not on matrices?
Would you consider adding a new method for it that works on matrices? Or a related function like `add_to_expression_matrix` which takes a tuple of dimensions as arguments? Or which finds the shape from the object?

Reply:
Hey Jacob @cfe316,
We've found that add_to_expression! has some unexpected behaviours when broadcast or used in the same manner as += , -=, etc. so using nested for-loops has been a lot safer.
We could make a wrapper function but I've found using a for-loop to often be clearer and safest